Battle/game.swift

```diff
@@ -116,3 +87,51 @@ class Game {
     
   
 }
+
+func newPersoWarrior () -> Warrior {
+    print("Warrior","\n choisis le nom de ton Warrior")
+    if let characterNamed = readLine(){
+        // on rajoute le nom choisi au personnage
+        let warrior = Warrior(name: characterNamed)
+        // on informe le joueur de son choix
+        print("\(characterNamed) is the Warrior")
+        return warrior
+    }
+    
+    // si characterNamed est nul on renvoi un Warrior sans nom
+   return Warrior(name:"")
+}
+
+func newPersoColosus() -> Colosus {
+
+print("Colosus",
+                "\n choisis le nom de ton colosus")
+    if let characterNamed = readLine(){
+        let colosus = Colosus(name: characterNamed)
+        print("\(characterNamed) is the Colosus")
+        return colosus
+    }
+    return Colosus(name: "")
+}
+
+func newPersoDwarf () -> Dwarf {
+    print("dwarf",
+                     "\n choisis le nom de ton Dwarf")
+        if let characterNamed = readLine(){
+            let dwarf = Dwarf(name: characterNamed)
+            print("\(characterNamed) is the Dwarf")
+    return dwarf
+}
+    return Dwarf(name:"")
+}
+
+func newPersoMagus ()->Magus {
+    print("magus",
+                    "\n choisis le nom de magus")
+        if let characterNamed = readLine(){
+            let magus = Magus(name: characterNamed)
+            print("\(characterNamed)is the Magus")
+        return magus
+        }
+    return Magus(name:"")
+}
```

Comment:
En fait la tu déplacé le code dans des méthodes mais sans forcéments mutaliser. 
Par exemple tu pourrais : 
```
func newCharacter (character: Character) -> Character {
    print("\(character.toString())","\n choisis le nom de ton \(character.toString())") // Tu crées une méthode toString ou toName que tu overrides dans chaque classe fille
    if let characterNamed = readLine(){
        // on rajoute le nom choisi au personnage
       character.name = characterNamed
        // on informe le joueur de son choix
        print("\(characterNamed) is the")
        return character
    }
```

C'est qu'un exemple et surement pas correct mais l'idée est d'avoir moins de ligne de code et de supprimer ce qui est redondant : 
-> print("magus", "\n choisis le nom de magus")
-> if let characterNamed = readLine(){
-> print("\(characterNamed)is the Magus")
 ->return magus
   -> return Magus(name:"")